src/views/monitor_history.view.php

```diff
@@ -45,13 +45,51 @@ public function setTotal(int $total): self {
         return $this;
     }
 
+    /**
+     * Build a list of history items with an optional interval string to the next (older) entry.
+     * The last item on the current page has no interval.
+     *
+     * @return array<int, array{entry:\Cronbeat\PingData, interval: ?string}>
+     */
+    public function getHistoryWithIntervals(): array {
+        $result = [];
+        $count = count($this->history);
+        for ($i = 0; $i < $count; $i++) {
+            $entry = $this->history[$i];
+            $next = $i + 1 < $count ? $this->history[$i + 1] : null;
+            $interval = $next !== null ? $this->computeIntervalString($entry, $next) : null;
+            $result[] = [
+                'entry' => $entry,
+                'interval' => $interval,
+            ];
+        }
+        return $result;
+    }
+
+    private function computeIntervalString(\Cronbeat\PingData $current, \Cronbeat\PingData $next): ?string {
+        try {
+            $t1 = new \DateTimeImmutable($current->getPingedAt());
+            $t2 = new \DateTimeImmutable($next->getPingedAt());
+            // Use DateTime::diff and DateInterval::format as requested in review
+            $diff = $t2->diff($t1);
+            $days = (int)$diff->format('%a');
+            $hours = (int)$diff->format('%H');
+            $minutes = (int)$diff->format('%I');
+            $seconds = (int)$diff->format('%S');
+            $totalHours = $days * 24 + $hours;
+            return sprintf('%02d:%02d:%02d', $totalHours, $minutes, $seconds);
```

Comment:
@jetbrains-junie just `return $diff->format...` and go by format `days:hours:minutes:seconds`